frontend/class-json-ld.php

```diff
@@ -0,0 +1,270 @@
+<?php
+/**
+ * @package WPSEO\Frontend
+ */
+
+/**
+ * Class WPSEO_JSON_LD
+ *
+ * Outputs schema code specific for Google's JSON LD stuff
+ *
+ * @since 1.8
+ */
+class WPSEO_JSON_LD {
+
+	/**
+	 * @var array Holds the plugins options.
+	 */
+	public $options = array();
+
+	/**
+	 * @var array Holds the social profiles for the entity
+	 */
+	private $profiles = array();
+
+	/**
+	 * @var array Holds the data to put out
+	 */
+	private $data = array();
+
+	/**
+	 * Class constructor
+	 */
+	public function __construct() {
+		$this->options = WPSEO_Options::get_all();
+
+		add_action( 'wpseo_head', array( $this, 'json_ld' ), 90 );
+		add_action( 'wpseo_json_ld', array( $this, 'website' ), 10 );
+		add_action( 'wpseo_json_ld', array( $this, 'organization_or_person' ), 20 );
+	}
+
+	/**
+	 * JSON LD output function that the functions for specific code can hook into
+	 *
+	 * @since 1.8
+	 */
+	public function json_ld() {
+		do_action( 'wpseo_json_ld' );
+	}
+
+	/**
+	 * Outputs the JSON LD code in a valid JSON+LD wrapper
+	 *
+	 * @since 1.8
+	 *
+	 * @param string $context The context of the output, useful for filtering
+	 */
+	protected function output( $context ) {
+		/**
+		 * Filter: 'wpseo_json_ld_output' - Allows filtering of the JSON+LD output
+		 *
+		 * @api array $output The output array, before its JSON encoded
+		 *
+		 * @param string $context The context of the output, useful to determine whether to filter or not
+		 */
+		$this->data = apply_filters( 'wpseo_json_ld_output', $this->data, $context );
+
+		if ( function_exists( 'wp_json_encode' ) ) {
+			$json_data = wp_json_encode( $this->data );  // wp_json_encode was introduced in WP 4.1
+		}
+		else {
+			$json_data = json_encode( $this->data );
+		}
+
+		if ( is_array( $this->data ) && ! empty( $this->data ) ) {
+			echo "<script type='application/ld+json'>", $json_data, '</script>', "\n";
+		}
+
+		// Empty the $data array so we don't output it twice.
+		$this->data = array();
+	}
+
+	/**
+	 * Schema for Organization
+	 */
+	private function organization() {
+		if ( '' !== $this->options['company_name'] ) {
+			$this->data['@type'] = 'Organization';
+			$this->data['name']  = $this->options['company_name'];
+			$this->data['logo']  = $this->options['company_logo'];
+			return;
+		}
+		$this->data = false;
+	}
+
+	/**
+	 * Schema for Person
+	 */
+	private function person() {
+		if ( '' !== $this->options['person_name'] ) {
+			$this->data['@type'] = 'Person';
+			$this->data['name']  = $this->options['person_name'];
+			return;
+		}
+		$this->data = false;
+	}
+
+	/**
+	 * Prepares the organization or person markup
+	 */
+	private function prepare_organization_person_markup() {
+		$this->fetch_social_profiles();
+
+		$this->data = array(
+			'@context' => 'http://schema.org',
+			'@type'    => '',
+			'url'      => WPSEO_Frontend::get_instance()->canonical( false, true ),
+			'sameAs'   => $this->profiles,
```

Comment:
If you return the profiles in the method you can call `$this->fetch_social_profiles();` directly